examples/nlp/language_modeling/megatron_change_num_partitions_pp.py

```diff
@@ -0,0 +1,753 @@
+                app_state.model_parallel_size = (
+                    app_state.pipeline_model_parallel_size * app_state.tensor_model_parallel_size
+                )
+
+                model = cls.restore_from(
+                    restore_path=args.model_file,
+                    trainer=trainer,
+                    map_location=torch.device("cpu"),
+                    save_restore_connector=NLPSaveRestoreConnector(),
+                )
+                model.to(dtype=dtype)
+
+                # Reset env flag
+                os.environ.pop(NEMO_MEGATRON_MODEL_PARALLEL_APPSTATE_OVERRIDE, None)
+
+                print(f"<<<<<<<< LOADED MODEL {pp_rank} {tp_rank} | GLOBAL RANK = {global_rank} >>>>>>>>>")
+                params = [p for _, p in model.named_parameters()]
+                partitions[pp_rank].append(params)
+
+                # app_state is being updated incorrectly during restore
+                app_state.data_parallel_rank = 0
+                app_state.pipeline_model_parallel_rank = pp_rank
+                app_state.tensor_model_parallel_rank = tp_rank
+                app_state.pipeline_model_parallel_size = pp_size
+                app_state.tensor_model_parallel_size = tp_size
+                app_state.model_parallel_size = (
+                    app_state.pipeline_model_parallel_size * app_state.tensor_model_parallel_size
+                )
+
+        # Build a unified model with PP 1 TP 1
+        model.cfg.tensor_model_parallel_size = 1
+        model.cfg.pipeline_model_parallel_size = 1
+        app_state.tensor_model_parallel_rank = 0
+        app_state.pipeline_model_parallel_size = 0
+        app_state.model_parallel_size = 1
+
+        trainer = Trainer(devices=1, strategy=NLPDDPStrategy(), accelerator="cpu", precision=precision)
+        if args.tokenizer_model_path is not None:
+            model.cfg.tokenizer.model = args.tokenizer_model_path
+
+        model = cls(model.cfg, trainer).to('cpu')
+        model._save_restore_connector = NLPSaveRestoreConnector()
+
+        if tgt_tp_size > 1 or tgt_pp_size > 1:
+            merge_partition(model, partitions)
+        else:
+            # Write out the PP 1 TP 1 model to disk
+            merge_partition(model, partitions, args.target_file)
+
+        # Empty cache memory of all parameters from all PP TP partitions
+        partitions.clear()
+
+    else:
+        # If input model has TP = 1 and PP = 1
+        app_state.model_parallel_size = 1
+        model = cls.restore_from(restore_path=args.model_file, trainer=trainer, map_location=torch.device("cpu"))
+        model.to(dtype=dtype)
+
+    # If target model has TP > 1 or PP > 1
+    if tgt_pp_size > 1 or tgt_tp_size > 1:
+
+        # Preserve the TP 1 PP 1 model parameters and names
+        global_params = []
+        global_params.append([p for n, p in model.named_parameters()])  # params
+        global_params.append([n for n, p in model.named_parameters()])  # names
+
+        if DEBUG_PRINT:
```

Comment:
use `logging.debug` instead of print and this `DEBUG_PRINT` condition

Reply:
Fixed all prints to debug / info / warning logging